### PR TITLE
fix: Remove grounding module related types

### PR DESCRIPTION
**Pull request by jjtang1985**
## Context

This PR contains:

- Remove grounding module related types.
- Update orchestration clients by picking non-breaking changes from #269 

## Definition of Done

- [x] Code is tested (Unit, E2E)
- [x] (Optional) Release notes updated

### CODE REVIEW

File: .changeset/remove-grounding.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+'@sap-ai-sdk/orchestration': minor
+---
+[Compatibility Note] Remove grounding module related types, which should not be used.
```
Comment (deekshas8) [inline .changeset/remove-grounding.md:4]: [q] Do we really need a compatibility note? This was never meant to be part of the public api. The fact that we expose `ModuleConfigs` in `index.ts` is incorrect.

File: packages/orchestration/src/client/api/schema/azure-content-safety-filter-config.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ import type { AzureContentSafety } from './azure-content-safety.js';
  */
 export type AzureContentSafetyFilterConfig = {
   /**
-   * String represents name of the filter provider.
+   * String represents name of the filter provider
```
Comment (deekshas8) [inline packages/orchestration/src/client/api/schema/azure-content-safety-filter-config.ts:12]: [pp] I think these changes are fine for now. But it's better to have proper linted code. I would like to adjust our eslint config to fix such issues (and only disable consistent-typedefinitions rule).

**Review by deekshas8 — APPROVED** [review]
LGTM. Please consider the comment about having the changeset.

File: .changeset/remove-grounding.md (hunk 1 above)
Comment (jjtang1985) [reply .changeset/remove-grounding.md]: For SDK user, I agree, it's not relevant, if they only use public APIs, so I deleted the compatibility notes.
I put it there, to make it transparent for those who noticed our grounding related code.

Comment (jjtang1985) [reply .changeset/remove-grounding.md]: will work on `ModuleConfigs` in a different PR.

**Comment by jjtang1985** [comment]
> Please consider the comment about having the changeset.

Removed already 😄

**Pull request merged**